tracer/src/Datadog.Trace/Iast/Analyzers/HardcodedSecretsAnalyzer.cs

```diff
@@ -94,12 +92,11 @@ private void PoolingThread()
                     }
                 }
 
-                _waitEvent.Wait(2_000);
+                await Task.Delay(2000, cancellationToken).ConfigureAwait(false);
             }
         }
-        catch (Exception err)
+        catch (Exception err) when (!(err is OperationCanceledException))

```

Comment:
I see you're letting the OperationCanceledException bubble up. I'm always confused with how cancellation behaves, have you checked that it doesn't trigger the `.ContinueWith(t => Log.Error(t.Exception, "Error in Hardcoded secret analyzer"), TaskContinuationOptions.OnlyOnFaulted)` continuation?

Reply:
The OnlyOnFaulted would be triggered if the status is set to Faulted. An OperationCanceledException should set the status to Canceled.